app/services/c100_app/address_lookup_service.rb

```diff
@@ -0,0 +1,64 @@
+module C100App
+  class AddressLookupService
+    ORDNANCE_SURVEY_URL = 'https://api.ordnancesurvey.co.uk/places/v1/addresses/postcode'.freeze
+
+    attr_accessor :postcode, :result
+
+    def initialize(postcode)
+      @postcode = postcode
+      @result = MapAddressLookupResults.call(results)
```

Comment:
I think if you remove this from the `#initialize`, it will be easier to test. So one step is initialising the service, and another step is calling `#results` to get the results (which, if we don't have them, will trigger the API call).

Otherwise there is a complex chain of events (init -> call API -> map results from API)